app/views/foreman/job_templates/update_packages_by_search_query_-_katello_ansible_default.erb

```diff
@@ -16,14 +16,14 @@ template_inputs:
   required: false
   value_type: plain
 %>
+<% if input('Packages search query').empty? && input('Selected update versions').empty? -%>
+<%= render_template('Package Action - Ansible Default', :state => 'latest', :name => '"*"') %>
+<% else -%>
 <% package_names = @host.package_names_for_job_template(
   action: 'update',
   search: input('Packages search query'),
   versions: input('Selected update versions')
 ) -%>
-<% if package_names.empty? -%>
-<%= render_template('Run Command - Ansible Default', :command => "yum -y update") %>
```

Comment:
I think it should be enough to replace line 25 with the code that you moved to line 20. 

There is the method here https://github.com/Katello/katello/blob/master/app/models/katello/concerns/host_managed_extensions.rb#L498 that takes care of the package search and it should return the correct packages based on the search query. For empty search query it simply returns an empty array. For yum packages it considers nvra, for deb packages we do not check the version so anything that you put into the version field for deb packages will be ignored.

Reply:
I have applied your feedback; this resulted in a much smaller diff.